osdu/client/_service_principal_util.py

```diff
@@ -116,5 +116,5 @@ def get_service_principal_token(self, resource_prefix):
             token_url, client_id, aws_oauth_custom_scope)
 
         response = requests.post(url=token_url, headers=headers)
-
-        return json.loads(response.content.decode())['access_token'], json.loads(response.content.decode())['expires_in'] + time()
+        response_json = json.loads(response.content.decode())
```

Comment:
I would still add the line right above this before trying to deserialize.
```
response.raise_for_status()
``` 
It should have already been there.
Also, please be sure to update the comment header at the top of this file with a log entry for these changes, since this file has been adopted from AWS under Apache license.

Reply:
It will throw a more relevant exception that way.

Reply:
And added a log entry!